haystack/components/audio/__init__.py

```diff
@@ -2,7 +2,16 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-from haystack.components.audio.whisper_local import LocalWhisperTranscriber
-from haystack.components.audio.whisper_remote import RemoteWhisperTranscriber
+import sys
+from typing import TYPE_CHECKING
 
-__all__ = ["LocalWhisperTranscriber", "RemoteWhisperTranscriber"]
+from lazy_imports import LazyImporter
+
+_import_structure = {"whisper_local": ["LocalWhisperTranscriber"], "whisper_remote": ["RemoteWhisperTranscriber"]}
+
+if TYPE_CHECKING:
+    from .whisper_local import LocalWhisperTranscriber
+    from .whisper_remote import RemoteWhisperTranscriber
+
+else:
+    sys.modules[__name__] = LazyImporter(name=__name__, module_file=__file__, import_structure=_import_structure)
```

Comment:
- I haven't found effective ways to reduce the import duplication (and ensure that IDE support works)
- In general, I am now using relative imports (`from .whisper_local ...`) instead of absolute ones because I think that having very similar expressions in `_import_structure` and real imports can help avoid errors.
- The code is copied from [lazy-imports README](https://github.com/telekom/lazy-imports?tab=readme-ov-file#usage--example-for-lazyimporter)